src/components/DynamicMenuItem/index.d.ts

```diff
@@ -0,0 +1,8 @@
+import { MenuItemProps } from '../MenuItem';
+
+export interface DynamicMenuItemProps extends Omit<MenuItemProps, 'disabled'> {
+    renderIf?: (rowData: Record<string, any>) => boolean;
+    disabled?: (rowData: Record<string, any>) => boolean;
```

Comment:
here let's pass an object instead: ```{ row }```, to leave it open to extend in the future, to pass index or whatever new thing we may need